codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -497,227 +598,147 @@ public boolean remove(T name) {
 
     @Override
     public Headers<T> clear() {
-        map.clear();
-        hasMultipleValues = false;
+        Arrays.fill(entries, null);
+        head.before = head.after = head;
         size = 0;
         return this;
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public Iterator<Entry<T, T>> iterator() {
-        Object iter = map.entrySet().iterator();
-        return !hasMultipleValues ? (Iterator<Entry<T, T>>) iter
-                                  : new NameValueIterator((Iterator<Entry<T, Object>>) iter);
+        return new HeaderIterator();
     }
 
     @Override
     public Boolean getBoolean(T name) {
         T v = get(name);
-        if (v == null) {
-            return null;
-        }
-        try {
-            return valueConverter.convertToBoolean(v);
-        } catch (Throwable ignored) {
-            return null;
-        }
+        return v != null ? valueConverter.convertToBoolean(v) : null;
     }
 
     @Override
     public boolean getBoolean(T name, boolean defaultValue) {
         Boolean v = getBoolean(name);
-        return v == null ? defaultValue : v;
+        return v != null ? v : defaultValue;
     }
 
     @Override
     public Byte getByte(T name) {
         T v = get(name);
-        if (v == null) {
-            return null;
-        }
-        try {
-            return valueConverter.convertToByte(v);
-        } catch (Throwable ignored) {
-            return null;
-        }
+        return v != null ? valueConverter.convertToByte(v) : null;
     }
 
     @Override
     public byte getByte(T name, byte defaultValue) {
         Byte v = getByte(name);
-        return v == null ? defaultValue : v;
+        return v != null ? v : defaultValue;
     }
 
     @Override
     public Character getChar(T name) {
         T v = get(name);
-        if (v == null) {
-            return null;
-        }
-        try {
-            return valueConverter.convertToChar(v);
-        } catch (Throwable ignored) {
-            return null;
-        }
+        return v != null ? valueConverter.convertToChar(v) : null;
     }
 
     @Override
     public char getChar(T name, char defaultValue) {
         Character v = getChar(name);
-        return v == null ? defaultValue : v;
+        return v != null ? v : defaultValue;
```

Comment:
To be consistent with the other methods that do value conversion.  I can change them all to `== null`, but do you think it matters?